### DISCUSSION TITLE
Next 12.x build - performance drop with cold starts

### DISCUSSION THREAD

**Discussion by petrvecera** [vercel/next.js#40577] [Help] [answered]
Hello guys

We run our app using AWS Lambda. Prior Next 12.x we were using "serverless" build target. This was deprecated with Next 12.x and "standalone" build is now the default one with serverless no longer supported. 

**However we noticed significant response time increase in cold starts - and also when lambdas are scaling (again cold start).** 

We tried to perf our code and found out that the line `const NextServer = require('next/dist/server/next-server').default` takes almost `~350ms` to execute. 

Our code to handle the requests on servless looks more or less like this 
```
// ...

// This line is heavy 
const NextServer = require('next/dist/server/next-server').default

const handle = new NextServer({
    hostname: 'localhost',
    port: port,
    dir: path.resolve(__dirname, '..'),
    dev: false,
    conf: config,
  }).getRequestHandler()

handle(req, res)
// ...
```

**We are wondering:**
- Is the next-server in the /dist/ folder correctly compiled? Tree Shaking and everything applied? 
- Are we running our server incorrectly? 
- Is there way to still use serverless build with Next 12?

**Comment by petrvecera** [comment]
Some of our projects can't be upgrade to Next 12 because they suffer from the slow cold starts. 

Anyone got any insight into this? Tips and so on ?

**Comment by petrvecera** [comment, marked as answer]
This issues has been resolved with Next 13

**Reply by stefanstankovic7189** [reply]
How?

**Reply by petrvecera** [reply]
When we updated to Next 13 we no longer see the long cold starts using the same code to run our server.
